internal/bgp/bgpfrr/config.go

```diff
@@ -0,0 +1,127 @@
+package bgpfrr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"text/template"
+)
+
+// TODO configFileName will need to be set to the filename of the FRR configuration
+// file in the container.
+var configFileName = "frr.conf"
+
+const configTemplate = `
+log stdout {{.Loglevel}}
+hostname {{.Hostname}}
+
+{{range .Routers -}}
+router bgp {{.Asn}}
+  no bgp default ipv4-unicast
+  bgp router-id {{.RouterId}}
+{{range .Neighbors }}
+  neighbor {{.Addr}} remote-as {{.Asn}}
+{{- end }}
+{{range $n := .Neighbors -}}
+{{range .Advertisements }}
+  address-family {{.Version}} unicast
+    neighbor {{$n.Addr}} activate
+    network {{.Prefix}}
+  exit-address-family
+{{- end}}
+{{end}}
+{{end}}
+`
+
+type frrConfig struct {
+	Loglevel string
+	Hostname string
+	Routers  map[string]*routerConfig
+}
+
+type routerConfig struct {
+	Asn       uint32
+	RouterId  string
+	Neighbors map[string]*neighborConfig
+}
+
+type neighborConfig struct {
+	Asn            uint32
+	Addr           string
+	Advertisements map[string]*advertisementConfig
```

Comment:
same here I think this is might be better off slice instead of map

Reply:
For this, I may be able to use a slice? Do you have a reason for the preference?

Reply:
If we are not accessing it by key but only iterating (and I think we are) I'd second @msherif1234 .
More than unnecessary hashing, it removes unnecessary complexity when reading the code.

Reply:
Yeah, I think that is fair for this one. For the others, I can remove the map if you think it would be better but it would mean iterating over lists to search for duplicates. wdyt?

Reply:
My opinion is that if we access the map by key (as you said, for finding duplicates), then it should remain a map

Reply:
In the process of prototyping using slices for advertisements, I seemed to come across a problem due to the non-deterministic ordering of an iteration over a map. That made me think that it may be better to uses slices in order to ensuring ordering. If the order changes, we will generate a new config file. I pushed this as a separate commit in the PR so let me know what you think.

Reply:
I am a bit torn about this. I think that if using a map is a better fit, we should be using a map and sort before rendering, even if it costs us more. I don't think the performance penalty is big enough to sacrifice readability.

Reply:
I think it's more readable in the sense the the sort would be isolated and for one specific purpouse. Again, I am fine with both solutions and just thinking out loudly, but I'd slightly prefer the map if we access it by key.